### ISSUE TITLE
`padding-line-between-statements`: also apply settings for some statements when prefixed with `export`

### ISSUE THREAD

**Issue by Brainshaker95** [eslint-stylistic/eslint-stylistic#539]
### Clear and concise description of the problem

Currently there are settings for all of the following statements in addition to `export`:

```
var
let
const
singleline-var
singleline-let
singleline-const
multiline-var
multiline-let
multiline-const
class
default
function
function-overload
enum
type
interface
```

Settings specified for those statements are not recognized anymore if they are part of an `export` statement.  
This might be by design or it might be a bug, depending on how you look at it (since there is a separate configuration for `export` statements as well). 

Assuming the following configuration...

```js
{
  '@stylistic/padding-line-between-statements': [
    'error',
    {
      blankLine: 'never',
      prev: ['singleline-const'],
      next: ['singleline-const'],
    },
  ],
}
``` 

...this piece of code would be correct:

```js
const a = 'a';
const b = 'b';
export const c = 'c';
                      // <- this is the line I am not expecting to be allowed
export const d = 'd';
```

Of course this specific case could be solved by just adding another object to the rule for `export` statements  but with those there is no possibility to differentiate between single and multiline statements.

### Suggested solution

I propose that settings given to the mentioned statements are also be applied when they are prefixed by an `export` keyword.  
My preferred solution would be that this is automatically handled and any rules specified for the `export` statment itself would take precedence over this implicit behavior.

### Alternative

What would also solve the issue (but what would also be way more implementation and maintenance effort) is to introduce new `StatementTypes` for all of the mentioned ones:

```
exported-var
exported-let
exported-const
exported-singleline-var
exported-singleline-let
exported-singleline-const
exported-multiline-var
exported-multiline-let
exported-multiline-const
exported-class
exported-default
exported-function
exported-function-overload
exported-enum
exported-type
exported-interface
```

I recognize that this would probably be a bad way moving forward but I still wanted to share the idea.  
I am open for any other suggestions so let me know if you any. I am happy to provide additional input if needed.

### Additional context

[This issue (#197)](https://github.com/eslint-stylistic/eslint-stylistic/issues/197) describes the same problem with a more specific scope only on functions.

### Validations

- [X] Follow our [Code of Conduct](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://eslint.style/contribute/guide).
- [X] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.

### Contributes

- [X] If this feature request is accepted, I am willing to submit a PR to fix this issue

**Comment by Brainshaker95** [comment]
I have only just noticed that this has basically already been implemented by https://github.com/eslint-stylistic/eslint-stylistic/pull/582.
See https://github.com/eslint-stylistic/eslint-stylistic/discussions/580 for the proposal. It does not allow for granular control on the different types of keywords that can be prefixed with `export` but that is a use case I thought of that I do not even have so I am going to close this issue.
